### PR TITLE
Check CSRF and Permissions while handling POST actions

### LINKED ISSUE #3: Post methods should check csrf
https://github.com/SpigotMC/XenforoApiKeys/blob/ed020375b3c111d08cd492c87fcf70d4e9390588/upload/library/XenforoApiKeys/ControllerPublic/Index.php#L26

Examples can be found by checking for "isConfirmedPost" usage in the XF codebase

### LINKED ISSUE #4: Post actions need to check permissions and also that user is logged in (user ID exists)
https://github.com/SpigotMC/XenforoApiKeys/blob/ed020375b3c111d08cd492c87fcf70d4e9390588/upload/library/XenforoApiKeys/ControllerPublic/Index.php#L31

### PR DESCRIPTION
**Pull request by jacobsandersen**
Closes #3 and closes #4.

Now checking CSRF and permissions (including checking user is logged in) while handling POST actions.

Update the main API key template to conditionally render based on the permissions the user has.

Update template to use configurable phrases.

### CODE REVIEW

File: upload/library/XenforoApiKeys/ControllerPublic/Index.php [hunk 1]
```diff
@@ -24,8 +34,22 @@ public function actionIndex() {
...
         $user = XenForo_Visitor::getInstance();
+        if (!$this->_doesUserExist($user['user_id'])) {
+          return $this->responseError(
+            "You must be logged in to perform that action.",
+            401
+          );
+        }
+
+        if (!$this->_checkPermission($user["permissions"], "obtain_api_key")) {
+          return $this->responseError(
```
Comment (md-5) [inline upload/library/XenforoApiKeys/ControllerPublic/Index.php]: $this->responseNoPermission() I think

Comment (jacobsandersen) [reply upload/library/XenforoApiKeys/ControllerPublic/Index.php]: Will take a look soon and change that

**Comment by jacobsandersen** [comment]
You were correct. Using responseNoPermission successfully triggers the XF standard "You must be logged in" view when the user is a guest. The new permission checking in the template does not render the buttons to send delete/rotate key requests when permission is revoked, however if they do somehow send those requests by being mischievous, the responseNoPermission() will fire there too.

**Comment by md-5** [comment]
lgtm

**Pull request merged**